ivadomed/adaptative.py

```diff
@@ -0,0 +1,695 @@
+import copy
+import os
+from os import path
+
+import h5py
+import numpy as np
+import pandas as pd
+from PIL import Image
+from bids_neuropoly import bids
+from medicaltorch import datasets as mt_datasets
+from tqdm import tqdm
+
+
+class Dataframe:
+    """
+    This class aims to create a dataset using an HDF5 file, which can be used by an adapative loader
+    to perform curriculum learning, Active Learning or any other strategy that needs to load samples in a specific way.
+    It works on RAM or on the fly and can be saved for later.
+    """
+
+    def __init__(self, hdf5, contrasts, path, target_suffix=None, roi_suffix=None,
+                 slices=False, dim=2):
```

Comment:
SOrry to be picky, could we rename `slices` by `slice_filter_fn`? I was not sure what it meant.